esp/esp/program/modules/handlers/programprintables.py

```diff
@@ -788,43 +789,89 @@ def getSchedule(program, user, schedule_type=None, room_numbers=True):
+                if len(times) == 0:
+                    # don't show classes with no times
+                    continue
                 else:
-                    rooms = u' ' + u", ".join(rooms)
+                    times = ' ' + ', '.join(times)
+                schedule += """
+                            <tr>
+                            <td> %s </td>
+                            <td> %s </td>
+                            """ % (str(times), cls.title())
                 if room_numbers:
-                    schedule += u'%s|%s|%s\n' % ((u' '+u",".join(cls.friendly_times())).ljust(24), (u' ' + cls.title()).ljust(40), rooms)
-                else:
-                    schedule += u'%s|%s\n' % ((u' '+u",".join(cls.friendly_times())).ljust(24), (u' ' + cls.title()).ljust(40))
+                    rooms = cls.prettyrooms()
+                    if len(rooms) == 0:
+                        rooms = 'N/A'
+                    else:
+                        rooms = ' ' + ', '.join(rooms)
+                    schedule += """
+                                <td> %s </td>
+                                """ % (str(rooms))
+                schedule += """
+                            </tr>
+                            """
+            schedule += """
+                        </indent>
+                        </table>
+                        """
 
         elif schedule_type == u'Volunteer':
-            schedule = u"""
-Volunteer schedule for %s:
-
- Time                   | Shift                                  \n""" % (user.name())
-            schedule += u'------------------------+----------------------------------------\n'
+            schedule = """
+                       <p> %s schedule for %s for %s: </p>
+                       """ % (schedule_type, user.name(), program.niceName())
+            schedule += """
+                        <table cellspacing="0" cellpadding="10" border="1" width="100%">
+                            <tr>
+                            <th width="35%">Time</th>
+                            <th width="65%">Shift</th>
+                            """
+            schedule += """
+                        </tr>
+                        <indent>
+                        """
             shifts = user.volunteeroffer_set.filter(request__program=program).order_by('request__timeslot__start')
             for shift in shifts:
-                schedule += u' %s| %s\n' % (shift.request.timeslot.pretty_time().ljust(23), shift.request.timeslot.description.ljust(39))
+                schedule += """
+                            <tr>
+                            <td> %s </td>
+                            <td> %s </td>
+                            </tr>
+                            """ % (str(shift.request.timeslot.pretty_time()), str(shift.request.timeslot.description))
+            schedule += """
+                        </indent>
+                    </table>
+                    """
+
+        return mark_safe(schedule)
```

Comment:
This mark_safe looks problematic because no HTML sanitization (escaping special symbols like <>, etc.) has been done when building up the schedule variable.  (Unless we're already sanitizing all names, classes, etc. before storing in the database?)  So it's not really safe to use directly in HTML output.

Instead, the schedule should be built up with format_html:

https://docs.djangoproject.com/en/1.8/ref/utils/#django.utils.html.format_html